backend/pkg/commons/db/statistics.go

```diff
@@ -841,34 +852,42 @@ func gatherValidatorDepositWithdrawals(day uint64, data []*types.ValidatorStatsT
 	depositsQry := `
 		with 
 		first_valid_deposits as (
-			select distinct on (publickey)
-				publickey,
-				block_slot,
-				block_index
+			select distinct on (blocks_deposits.publickey)
+				blocks_deposits.publickey,
+				blocks_deposits.block_slot,
+				blocks_deposits.block_index
 			from
 				blocks_deposits
+			inner join blocks on
+				blocks_deposits.block_root = blocks.blockroot and (blocks.status = '1' or blocks.slot = 0)
 			where
-				valid_signature
-				and publickey in (select bd.publickey from blocks_deposits bd where bd.valid_signature and bd.block_slot >= $1 and bd.block_slot <= $2)
+				-- filter out relevant requests for performance-reasons, no need to check for validity
+				blocks_deposits.publickey in (select bd.publickey from blocks_deposits bd where bd.block_slot >= $1 and bd.block_slot <= $2)
+				and blocks_deposits.valid_signature
 			order by
-				publickey,
-				block_slot,
-				block_index
+				blocks_deposits.publickey,
+				blocks_deposits.block_slot,
+				blocks_deposits.block_index
 		),
 		first_valid_deposit_requests as (
 			select
-				distinct on (pubkey)
-				pubkey,
-				block_slot,
-				request_index
+				distinct on (blocks_deposit_requests_v2.pubkey)
+				blocks_deposit_requests_v2.pubkey,
+				blocks_deposit_requests_v2.slot_processed as block_slot,
+				blocks_deposit_requests_v2.index_processed as request_index
 			from
-				blocks_deposit_requests
+				blocks_deposit_requests_v2
+			inner join blocks on
+				blocks_deposit_requests_v2.block_processed_root = blocks.blockroot and (blocks.status = '1' or blocks.slot = 0)
```

Comment:
this check may be redundant since we check for `blocks_deposit_requests_v2.status = 'completed'` anyway, but it doesnt hurt too much perf-wise and is good to be more explicit i guess